sdk/core/System.ClientModel/api/System.ClientModel.net6.0.cs

```diff
@@ -77,6 +77,32 @@ protected AsyncCollectionResult() { }
         public abstract System.ClientModel.ContinuationToken? GetContinuationToken(System.ClientModel.ClientResult page);
         public abstract System.Collections.Generic.IAsyncEnumerable<System.ClientModel.ClientResult> GetRawPagesAsync();
     }
+    public enum ClientAuthenticationMethod
+    {
+        EntraId = 0,
+        ApiKey = 1,
+        Subclient = 2,
+    }
+    public partial class ClientCache
+    {
+        public ClientCache() { }
+        public T Get<T>(System.Func<T> value, string id = "") where T : class { throw null; }
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct ClientConnection
+    {
+        private readonly object _dummy;
+        private readonly int _dummyPrimitive;
+        public ClientConnection(string id, string locator, object credential) { throw null; }
+        public ClientConnection(string id, string locator, string apiKey) { throw null; }
```

Comment:
Which ctor should be called when ClientAuthenticationMethod is Subclient? ... oh, and how do I get the Authentication method to be Subclient?

Reply:
What do you think about adding a new constructor for Subclient like this?

```chsarp
public ClientConnection(string id, string locator)
{
    if (string.IsNullOrWhiteSpace(id))
        throw new ArgumentException("Id cannot be null or empty.", nameof(id));
    if (string.IsNullOrWhiteSpace(locator))
        throw new ArgumentException("Locator cannot be null or empty.", nameof(locator));

    Id = id;
    Locator = locator;
    Authentication = ClientAuthenticationMethod.Subclient;
}
```

Reply:
@KrzysztofCwalina What do you think about adding a new constructor for Subclient like this?

```chsarp
public ClientConnection(string id, string locator)
{
    if (string.IsNullOrWhiteSpace(id))
        throw new ArgumentException("Id cannot be null or empty.", nameof(id));
    if (string.IsNullOrWhiteSpace(locator))
        throw new ArgumentException("Locator cannot be null or empty.", nameof(locator));

    Id = id;
    Locator = locator;
    Authentication = ClientAuthenticationMethod.Subclient;
}
```

Reply:
Added a new ctor public ClientConnection(string id, string locator) for NoAuth